### PR TITLE
Added Brier score for multiclass problems

### PR DESCRIPTION
**Pull request by PhilippPro**
(maybe the model.matrix( ~ . -1, data = as.data.frame(truth)))^2) could be done better, needs reshape2, which is imported in mlr)

### CODE REVIEW

**Comment by PhilippPro** [comment]
If it is ensured, that probabilities sum up to 1, the worst possible value would be 2. Is this ensured in mlr?

**Comment by jakob-r** [comment]
It would be great if you can add a test in `test_base_measures.R` in the part `"check measure calculations"`

**Comment by PhilippPro** [comment]
maybe let's first discuss, if to implement a new multiclass learner or implement this in the normal brier function, like discussed in #855

**Comment by catastrophic-failure** [comment]
I see no problem with Brier Score assuming different values in multiclass or binary tasks. You could scale it to [0,1] or let it be after all we want to compare either ranks (doesn't depend on scale) or performance within a task (hypothesis testing) in a benchmark experiment.

Besides the original Brier Score definition is the one you coded, which adds to 2 with any number of classes.

Looking at this this way, it would make sense to have two measures: you can use `brier` when your tasks are all binary, but if you include multiclass tasks among them `multiclass.brier` will work.

File: tests/testthat/test_base_measures.R [hunk 1]
```diff
@@ -245,6 +245,11 @@ test_that("check measure calculations", {
   acc.perf = performance(pred.classif, measures = acc, model = mod.classif)
   expect_equal(acc.test, acc$fun(pred = pred.classif))
   expect_equal(acc.test, as.numeric(acc.perf))
+  #multiclass.brier
+  multiclass.brier.test = mean(rowSums((getPredictionProbabilities(pred.classif) - matrix(c(0,0,1,0,1,0,0,1,0,1,0,0), 4, 3))^2))
+  multiclass.brier.perf = performance(pred.classif, measures = multiclass.brier, model = mod.classif)
+  expect_equal(multiclass.brier.test, multiclass.brier$fun(pred = pred.classif))
+  expect_equal(multiclass.brier.test, as.numeric(multiclass.brier.test))
```
Comment (catastrophic-failure) [inline tests/testthat/test_base_measures.R]: Should read `expect_equal(multiclass.brier.test, as.numeric(multiclass.brier.perf))` no?

**Comment by PhilippPro** [comment]
Somebody can tell me, why the term probabilites instead of probabilities is used in some functions?

**Comment by berndbischl** [comment]
> Somebody can tell me, why the term probabilites instead of probabilities is used in some functions?

because it is a typo i would guess?

**Pull request merged**